pkgs/development/libraries/webkitgtk/2.14.nix

```diff
@@ -37,24 +61,40 @@ stdenv.mkDerivation rec {
   cmakeFlags = [
   "-DPORT=GTK"
   "-DUSE_LIBHYPHEN=0"
-  "-DENABLE_GLES2=ON"
-  ];
+  ] ++ (if stdenv.isDarwin then [
+  "-DUSE_SYSTEM_MALLOC=ON"
+  "-DUSE_ACCELERATE=0"
+  "-DENABLE_INTROSPECTION=ON"
+  "-DENABLE_MINIBROWSER=OFF"
+  "-DENABLE_PLUGIN_PROCESS_GTK2=OFF"
+  "-DENABLE_MINIBROWSER=OFF"
+  "-DENABLE_VIDEO=ON"
+  "-DENABLE_QUARTZ_TARGET=ON"
+  "-DENABLE_X11_TARGET=OFF"
+  "-DENABLE_OPENGL=OFF"
+  "-DENABLE_WEB_AUDIO=OFF"
+  "-DENABLE_WEBGL=OFF"
+  "-DENABLE_GRAPHICS_CONTEXT_3D=OFF"
+  "-DENABLE_GTKDOC=OFF"
+  ] else [ "-DENABLE_GLES2=ON" ]);
 
   # XXX: WebKit2 missing include path for gst-plugins-base.
   # Filled: https://bugs.webkit.org/show_bug.cgi?id=148894
-  NIX_CFLAGS_COMPILE = "-I${gst-plugins-base.dev}/include/gstreamer-1.0";
+  NIX_CFLAGS_COMPILE = "-I${gst-plugins-base.dev}/include/gstreamer-1.0"
+                     + (optionalString stdenv.isDarwin " -lintl");
 
   nativeBuildInputs = [
     cmake perl python2 ruby bison gperf sqlite
     pkgconfig gettext gobjectIntrospection
   ];
 
-  buildInputs = [
-    gtk2 wayland libwebp enchant libnotify gnutls pcre nettle libidn
+  buildInputs = libintlOrEmpty ++ [
+    gtk2 libwebp enchant libnotify gnutls pcre nettle libidn
     libxml2 libsecret libxslt harfbuzz libpthreadstubs libtasn1 p11_kit
     gst-plugins-base libxkbcommon epoxy at_spi2_core
   ] ++ optional enableGeoLocation geoclue2
-    ++ (with xlibs; [ libXdmcp libXt libXtst ]);
+    ++ (with xlibs; [ libXdmcp libXt libXtst ])
+    ++ (if stdenv.isDarwin then [ libedit readline mesa ] else [ wayland ]);
```

Comment:
As above use `stdenv.lib.optionals` and make the platform explicit for `wayland`.